scipy/stats/tests/test_distributions.py

```diff
@@ -2601,16 +2604,13 @@ def test_sf(self):
         expected = (scale/x)**b   # 2.25e-18
         assert_allclose(p, expected)
 
-    @pytest.fixture(scope='function')
```

Comment:
I had not used a fixture like this before but just started to. We can't do this?

Reply:
No, this is not thread-safe. It would create `rng` only once per test, and then draw samples from it multiple times when run under `pytest-run-parallel`. So the random numbers change in different threads, making the test fail when tolerances are set tightly.

It also looks like a micro-optimization that makes the code harder to understand, so I would avoid such constructs.

Reply:
If we really need a fixture like this for something, it can still be used - but then the test itself should be marked with the `thread_unsafe` marker.